src/rpc/blockchain.cpp

```diff
@@ -1339,19 +1321,10 @@ UniValue getmintsinblocks(const UniValue& params, bool fHelp) {
                 HelpExampleCli("getmintsinblocks", "1200000 1000 5") +
                 HelpExampleRpc("getmintsinblocks", "1200000, 1000, 5"));
 
-    int nBestHeight = chainActive.Height();
-
-    int heightStart = params[0].get_int();
-    if (heightStart < Params().Zerocoin_StartHeight())
-        heightStart = Params().Zerocoin_StartHeight();
-
-    int range = params[1].get_int();
-    if (range < 1)
-        throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid block range. Must be strictly positive.");
+    LOCK(cs_main);
```

Comment:
Adding a small lock in `validaterange()` makes more sense and allows to remove this one over the whole function and more.